src/app/util/ember-compatibility-functions.cpp

```diff
@@ -253,6 +253,13 @@ CHIP_ERROR ReadSingleClusterData(FabricIndex aAccessingFabricIndex, const Concre
         ReturnErrorOnFailure(apWriter->Put(TLV::ContextTag(AttributeDataElement::kCsTag_Data), uint16_data));
         break;
     }
+    case ZCL_INT24U_ATTRIBUTE_TYPE: // Unsigned 24-bit integer
+    {
+        uint32_t uint32_data;
+        uint32_data = emberAfGetInt24u(attributeData, 0, 3);
```

Comment:
So this is assuming `attributeData` is little-endian, not native-endian, right?  Is that a correct assumption for the way the values are stored, especially the default values?

Reply:
This is only copying 3 bytes which are to be in the machine endianness (little or big). I believe that the attribute-data in the attribute store is always stored in machine (CPU) endianness.

Reply:
`emberAfGetInt24u` always reads little-endian data, no?